### ISSUE TITLE
CONNECTIVITY_CHANGE broadcast not received in apps targeting SDK 24+

### ISSUE THREAD

**Issue by weisers** [launchdarkly/android-client#52]
While testing changes to my app from updating the `targetSdkVersion` to 24 and above I noticed that the `ConnectivityReceiver` no longer receives `CONNECTIVITY_CHANGE` broadcasts.  From https://developer.android.com/about/versions/nougat/android-7.0-changes#bg-opt:

> Apps targeting Android 7.0 (API level 24) and higher do not receive CONNECTIVITY_ACTION broadcasts if they declare their broadcast receiver in the manifest. Apps will still receive CONNECTIVITY_ACTION broadcasts if they register their BroadcastReceiver with Context.registerReceiver() and that context is still valid.

The same issue affects the example app included in the repo as well.

We can get around it by registering the receiver manually within the app:

```java
IntentFilter intentFilter = new IntentFilter();
intentFilter.addAction(ConnectivityManager.CONNECTIVITY_ACTION);
context.registerReceiver(new ConnectivityReceiver(), intentFilter);
```

but wanted to know if there was a better, prefered way to do this.

**Comment by arun251** [comment]
We are looking into this.

**Comment by arun251** [comment]
Hi @weisers, we've released a fix for this in version 2.5.4. Thanks for the report.
